src/neptune/versionHelper.d

```diff
@@ -0,0 +1,334 @@
+/*******************************************************************************
+
+    Helper to deal with version strings
+
+    Copyright:
+        Copyright (c) 2017 Sociomantic Labs GmbH. All rights reserved.
+
+    License:
+        Boost Software License Version 1.0. See LICENSE.txt for details.
+
+*******************************************************************************/
+
+module neptune.versionHelper;
+
+import std.algorithm.iteration : splitter;
+import octod.api.repos : Tag;
+
+/// Structure to easily deal with version strings
+struct Version
+{
+    import std.typecons : Nullable;
+
+    /// Type of the version
+    enum Type { Major, Minor, Patch, Invalid };
+
+    /// Type of this version
+    Type type;
+
+    /// Major, minor and patch version numbers
+    Nullable!int major, minor, patch;
+
+    /// Any extra letters to the version (e.g. -preview, -breaking)
+    string extra;
+
+
+    /***************************************************************************
+
+        Constructor to make a version struct from the major, minor and patch
+        values
+
+        Params:
+            type = type of the version
+            major = major version
+            minor = minor version, optional
+            patch = patch version, optional
+            extra = any extra string following the version
+
+    ***************************************************************************/
+
+    this ( Type type, int major, int minor = -1, int patch = -1,
+           string extra = "" )
+    {
+        this.type = type;
+        this.major = major;
+
+        if (minor >= 0)
+            this.minor = minor;
+
+        if (patch >= 0)
+            this.patch = patch;
+
+        this.extra = extra;
+    }
+
+
+    /***************************************************************************
+
+        Constructor to make a version struct from a string
+
+        Params:
+            ver = version string (e.g. "v1.4.2" or "v1.x.x")
+
+    ***************************************************************************/
+
+    this ( string ver )
```

Comment:
This unified branch name parsing and actual version parsing in one utility. I see that it may have been useful in your case but it is incorrect from SemVer point of view (`v1.x.x` is not a valid version identifier per SemVer). Relevant code needs to be separated even if it requires some copy pasting.